sparkly/session.py

```diff
@@ -72,15 +74,33 @@ class MySession(sparkly.SparklySession):
             Key - a name of the function,
             Value - either a class name imported from a JAR file
                 or a tuple with python function and its return type.
+        app_id_tempate (str|None): if set and nonempty, generate the `spark.app.id` with
+            this template. Interpolation is available with some pre-defined variables:
+                * process_time: the time that the process started
+                * process_uid: a unique id associated with the process
+                * session_time: the time the session started
+                * session_uid: a unique id associated with the session
+            A default value is provided using the process-uid and session-time.
+            This helps a specific use case when running in Kubernetes: when a session
+            is restarted, the same app-id is used, breaking storage of spark-history data
+            (only the first session will have its history stored, unless overwrite mode
+            is used, in which case only the last session will have its history stored).
+            By defaulting to using the process-uid and session-time information, we get
+            sane "grouping" of all sessions originating from the same process, but also
+            achieve separate individual app ids so that history for each can be maintained.
+            To disable this functionality entirely, simply set to None or emptystring.
     """
     options = {}
     packages = []
     jars = []
     udfs = {}
     repositories = []
+    app_id_template = 'sparkly-{process_uid}-{session_time}'
 
     _instantiated_session = None
     _original_environment = None
+    _process_time = int(time.time())
```

Comment:
I would suggest that we set these to `None` here, and then in `__init__` we can do something like:
```
if self._process_time is None:
    self._process_time = int(time.time())
```
This way we can test the template behavior by mocking out `time` and `uuid` as usual.

This changes the definition of this attribute to "first instantiation time of the session", which I think is fine for our purposes (since `process_time` wasn't accurate either way, it was more first import time).